codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -401,6 +403,18 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
                                 "Multiple content-length headers received");
                     }
                 }
+            } else {
+                // Need to check trailers don't contain pseudo headers. According to RFC 9113
+                // Trailers MUST NOT include pseudo-header fields (Section 8.3).
+                if (!headers.isEmpty()) {
```

Comment:
Ideally, this logic should be inside `DefaultHttp2HeadersDecoder` or `HpackDecoder` where other validation is already in place. However, I don't see how those decoders can distinguish between headers and trailers. If someone has ideas, please share.

If we keep it here bcz of `isTrailers` flag, this logic should be enabled only when `validateHeaders/isValidateHeaders` flag is set to `true` (propagated from `AbstractHttp2ConnectionHandlerBuilder`).

Reply:
like you I thought it could have been in `Http2Headers`, but the creation of such instances is lacking of proper context to make this possible

Reply:
@isaacrivriv can you address comments of @idelpivnitskiy related to `validateHeaders` flag ?

Reply:
Yes I can make that change today, though since the RFC establishes this as a MUST, does anyone know if the `validateHeaders` flag default to true on every connection decoder?

Reply:
Pushed out a commit for this. I originally tried to add the validation by adding a flag to the DefaultHttp2ConnectionDecoder but unfortunately ran into an issue about changing APIs while building. I went through the process of adding a setter for the validate headers flag called when the builders create the decoder and set the flag to true as default. Any feedback/recommendations?